docs/how-to/hip_runtime_api/error_handling.rst

```diff
@@ -0,0 +1,126 @@
+.. meta::
+   :description: Error Handling
+   :keywords: AMD, ROCm, HIP, error handling, error
+
+********************************************************************************
+Error handling
+********************************************************************************
+
+HIP provides functionality to detect, report, and manage errors that occur
+during the execution of HIP runtime functions or when launching kernels. Every
+HIP runtime function, apart from launching kernels, has :cpp:type:`hipError_t`
+as return type. :cpp:func:`hipGetLastError()` and :cpp:func:`hipPeekAtLastError()`
+can be used for catching errors from kernel launches, as kernel launches don't
+return an error directly. HIP maintains an internal state, that includes the
+last error code. hipGetLastError returns and resets that error to hipSuccess,
+while :cpp:func:`hipPeekAtLastError` just returns the error without changing it.
+To get a human readable version of the errors, :cpp:func:`hipGetErrorString()`
+and :cpp:func:`hipGetErrorName()` can be used.
+
```

Comment:
Currently, there is a slight difference in the behavior of hipGetLastError when compared with cudaGetLastError.  
- hipGetLastError returns the returned error code of just the last API even if its hipSuccess.
- cudaGetLastError returns the error returned by any of the preceeding CUDA APIs in the same host thread. 

HIP behavior will be matched with CUDA in the next major ROCm release 7.0.  Meanwhile we are planning to introduce an environment variable in 6.4, which when enabled will match the CUDA behavior, otherwise it will continue with the current behavior by default.

Please check if we can mention this difference of the API behavior in this document.

Reply:
Added note.